src/components/validations/with-validation.hoc.js

```diff
@@ -12,36 +13,29 @@ const withValidation = (WrappedComponent) => {
       children: PropTypes.node, // Children elements
       name: PropTypes.string.isRequired, // Name to uniquely identify the component
       value: PropTypes.string, // The current value of the component
-      validations: PropTypes.oneOfType([ // The error validations that should be run against the value
-        PropTypes.func,
-        PropTypes.arrayOf(PropTypes.func)
-      ]),
-      warnings: PropTypes.oneOfType([ // The warnings validations that should be run against the value
-        PropTypes.func,
-        PropTypes.arrayOf(PropTypes.func)
-      ]),
-      info: PropTypes.oneOfType([ // The info validations that should be run against the value
-        PropTypes.func,
-        PropTypes.arrayOf(PropTypes.func)
-      ]),
       onBlur: PropTypes.func, // Custom function to be called when the component blurs
-      onChange: PropTypes.func // Custom function called when component value changes
+      onChange: PropTypes.func, // Custom function called when component value changes
+      ...Object.values(VALIDATION_TYPES).reduce((acc, type) => ({
+        ...acc,
+        [type]: PropTypes.oneOfType([ // The info validations that should be run against the value
+          PropTypes.func,
+          PropTypes.arrayOf(PropTypes.func)
+        ])
+      }), {})
```

Comment:
I liked the previous solution better. I think that we shouldn't compromise readability for a few lines of code less.

Reply:
It's not to reduce the lines of code, it is to try and address that we are supporting different kinds of validations and that code is spread across multiple components. Basing it from the same config allows us to better manage that going forward. One of the issues we had previously was trying to remember to update the code for all 3 variations.

Reply:
I agree that the previous code was more readable, but tools such as storybook means we shouldn't have developers looking at the code to see the defined prop types.

Reply:
I also agree it was a little more readable but doing it this way should make it easier if we ever add or remove validation types

Reply:
sounds reasonable :)